File: LeetCode_ProblemSet/Problem3396.cpp
#include <unordered_set>
#include <vector>

class Solution {
   public:
    int minimumOperations(std::vector<int>& nums) {
        int n = nums.size();
        std::unordered_set<int> set;
        for (int i = n - 1; i >= 0; i--) {
            if (set.find(nums[i]) == set.end())
                set.insert(nums[i]);
            else
                return std::ceil(float(i + 1) / 3);
        }
        return 0;
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem3396.rs
use std::collections::HashSet;

impl Solution {
    pub fn minimum_operations(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut set = HashSet::new();
        for i in (0..n).rev() {
            if !set.contains(&nums[i]) {
                set.insert(nums[i]);
            } else {
                return ((i + 1) as f32 / 3.0).ceil() as i32;
            }
        }
        return 0;
    }
}